src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php

```diff
@@ -78,9 +84,17 @@ public function createLivePropMetadata(string $className, string $propertyName,
             }
         }
 
-        $type = $property->getType();
-        if ($type instanceof \ReflectionUnionType || $type instanceof \ReflectionIntersectionType) {
-            throw new \LogicException(sprintf('Union or intersection types are not supported for LiveProps. You may want to change the type of property %s in %s.', $property->getName(), $property->getDeclaringClass()->getName()));
+        if (null === $type && null === $collectionValueType && isset($infoTypes[0])) {
```

Comment:
Is it possible that `$type` is null, but we DO have a `$collectionValueType`? Maybe gave `/** @var ChildDto[] */` but no `array` type on the field?

Reply:
But in this case we don't care, do we? `/** @var ChildDto[] */` is covered by tests already :)